src/pages/Home.js

```diff
@@ -2,37 +2,65 @@ import React, { useEffect, useState } from 'react';
 import ItemList from '../components/ItemList';
 import styled from 'styled-components';
 import ContentWrapper from '../styles/contentWrapper';
-// import mockData from '../db/mock_data.json';
 import axios from 'axios';
+import SearchField from '../components/Search/SearchField';
+import Pagination from '../components/Pagination/Pagination';
+// import mockData from '../db/mock_data.json';
 
 const API_URL =
   'https://gp-super-store-api.herokuapp.com/item/list?size=6&sortDir=asc';
 
 const StyledContainer = styled(ContentWrapper)`
   && {
+    display: flex;
     align-items: center;
-
+    flex-direction: column;
+    // justify-content: space-between;
     width: 100%;
     height: 100%;
-    display: flex;
-    flex-direction: column;
-    justify-content: space-between;
   }
 `;
 /* use this to feed with mockData */
 // const data = { products: mockData.items.slice(0, 6) };
 
 const Home = () => {
   const [productsData, setProductsData] = useState([]);
+  const [totalItems, setTotalItems] = useState(0);
+  const [hasMore, setHasMore] = useState(false);
   const [isFetching, setIsFetching] = useState(false);
+  const [search, setSearch] = useState('');
+  const [fromItem, setFromItem] = useState('');
+  const [page, setPage] = useState(0);
+
+  const onSearch = query => {
+    console.log({ query });
+    onPageChange(0);
+    setSearch(`&q=${query}`);
+  };
+
+  const onPageChange = pageNum => {
+    console.log({ pageNum });
+    // check if already in last page
+    if (pageNum * 6 >= totalItems && !hasMore) return;
```

Comment:
Try not to use "magic numbers" in your code. It's best to declare reoccurring numbers like this as constants somewhere in your project so if you need to change the value, you can just change the value of the constant.

Reply:
A "magic number" is any constant number that isn't 1 or 0.

Reply:
noted!